load_balancer/test/integration/load_balancerTest.js

```diff
@@ -0,0 +1,243 @@
+const lbUrl = `https://${configData.load_balancer.hostname}:${configData.load_balancer.port}`;
+const sandbox = sinon.createSandbox();
+
+let loadBalancer;
+let logStub;
+
+check('LBCONFIG');
+
+chai.should();
+chai.use(dirtyChai);
+
+const activateNocks = function activateNocks() {
+  if (!nock.isActive()) {
+    nock.activate();
+  }
+};
+
+const cleanNocks = function cleanNocks() {
+  nock.cleanAll();
+  nock.restore();
+};
+
+const stubConsole = function stubConsole() {
+  logStub = sandbox.stub(console, 'log');
+};
+
+const restoreConsole = function restoreConsole() {
+  logStub = {};
+  sandbox.restore();
+};
+
+const startLoadBalancer = function startLoadBalancer() {
+  delete require.cache[require.resolve('../../load_balancer.js')];
+  // eslint-disable-next-line global-require
+  loadBalancer = require('../../load_balancer.js');
+};
+
+const connectionCheckNock = function connectionCheckNock(url, returnVal) {
+  const nockObj = nock(url)
+    .get('/connectionCheck')
+    .reply(200, (uri) => returnVal);
+  return nockObj;
+};
+
+const requestRunNock = function requestRunNock(url) {
+  const tempJob = testData.submissionsData[0];
+  const nockObj = nock(url)
+    .persist()
+    .post('/requestRun')
+    .reply(200, (uri, requestBody) => {
+      requestBody.should.have.all.keys('id_no', 'Lab_No', 'time', 'commit', 'status', 'penalty', 'socket', 'language');
+      requestBody.should.deep.equal(tempJob);
+      return true;
+    });
+  return nockObj;
+};
+
+const connectionCheckAssert = function connectionCheckAssert(upArray, resp) {
+  request.get(`${lbUrl}/connectionCheck`, (error, response) => {
+    const responseBody = JSON.parse(response.body);
+    responseBody.components.forEach((elem, index) => {
+      if (upArray.includes(index)) {
+        elem.status.should.equal('up');
+      } else {
+        elem.status.should.equal('down');
+      }
+    });
+    restoreConsole();
+    resp();
+  });
+};
+
+const submitAssert = function submitAssert() {
+  const tempJob = testData.submissionsData[0];
+  request.post(`${lbUrl}/submit`, { json: tempJob }, (error, response) => {
+    (error === null).should.be.true();
+    response.body.should.equal(true);
+  });
+};
+
+describe('Correctly maintains list of ENs', () => {
+  beforeEach(() => {
+    stubConsole();
+    startLoadBalancer();
+    restoreConsole();
+    activateNocks();
+  });
+
+  afterEach((done) => {
+    cleanNocks();
+    loadBalancer.server.close(done);
+  });
+
+  it('during status check', (done) => {
+    stubConsole();
+    const res = JSON.parse(JSON.stringify(configData.Nodes));
+    const lbTestStatus = configData.load_balancer;
+    lbTestStatus.status = 'up';
+    [0,1].forEach((elem) => {
+      connectionCheckNock(`https://${res[elem].hostname}:${res[elem].port}`, true);
+      connectionCheckNock(`https://${res[elem].hostname}:${res[elem].port}`, true);
+    })
+    request.get(`${lbUrl}/connectionCheck`, (error, response) => {
```

Comment:
The coding conventions in this project prefer async/await over callbacks. Please change the call backs to async / await.

Reply:
Sir, I think that is the syntax of the request module. I'm not entirely sure, I've just used it in a similar way as Ankshit did in his execute_nodeTest.js

Reply:
Fine. Let the code remain.